types/s3-zip/index.d.ts

```diff
@@ -0,0 +1,30 @@
+import { S3Client } from "@aws-sdk/client-s3";
+import { Archiver, ArchiverOptions, EntryData, Format } from "archiver";
+
+export interface ArchiveOptions {
+    s3?: S3Client;
+    region?: string;
+    bucket: string;
+    debug?: boolean;
+    preserveFolderStructure?: boolean;
+}
+
+interface S3Zip {
+    archive(
+        opts: ArchiveOptions,
+        folder: string | undefined,
+        filesS3: string[],
+        filesZip?: EntryData[],
+    ): Archiver;
+
+    setFormat(format: Format): this;
+
+    setArchiverOptions(archiverOpts: ArchiverOptions): this;
+
+    // eslint-disable-next-line @typescript-eslint/ban-types -- Function
+    setRegisterFormatOptions(registerFormat: string, formatModule: Function): this;
+}
+
+declare const s3Zip: S3Zip;
+
+export default s3Zip;
```

Comment:
This should be `export = s3Zip`, as upstream does `module.exports = s3Zip`.

That or, just export each of these functions; I don't think you actually need to have an object when the top level only exports named functions.

Reply:
Hi Jake, I have updated the export to use export = s3Zip. I do believe it needs to be an object rather than exporting seperate functions otherwise I lose the chainability of the functions e.g.
```ts
// s3-zip-tests.ts
function testChaining(): void {
    s3Zip
        .setFormat("zip")
        .setArchiverOptions({ zlib: { level: 9 } })
        .setRegisterFormatOptions("zip", require("fs"))
        .archive({ region: region, bucket: bucket }, folder, [file1, file2, file3, file4]);
}
```

Reply:
Changing to the use of `export =` syntax means I have lost the ability to export the `ArchiveOptions` interface as well. While this isn't a requirement (just more of a nice bonus incase people wanted to type their options in a separate object outside the call to the archiver function) if you know of anyway to export this as well that would be great. Tying to export both gives the message 

> An export assignment cannot be used in a module with other exported elements.ts(2309)

However, from a bit of searching I don't think this is possible and I am happy with where this PR is at and would like to get it merged asap.

Thanks again.

Reply:
Ah, I see. You can declare a namespace with the same name as the object and put the extra type in there, the export equals that

Reply:
Wonderful, thank you @jakebailey! I have updated the PR and believe it should now be good to go